docs/usage/apiserver-sni-injection.md

```diff
@@ -0,0 +1,55 @@
+# APIServerSNI environment variable injection
```

Comment:
Can you add the document to the [index](https://github.com/gardener/gardener/blob/master/docs/README.md)?